src/memetic_algorithm.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
```

Comment:
Basically all the same comments in this script as the other. In other words, address the issues brought up above here too.